### PR TITLE
adds /api/articles(topic query) with TDD, MVC and error handling

### PR DESCRIPTION
**Pull request by EliR94**
Hi, here is the code for 11-get-articles-by-topic-query

This includes:

- tests for /api/articles(topic query)
- code within articles and topics MVC to allow /api/articles to take a topic query and send data (filtered by topic) back to the client
- error handling for /api/articles(topic query)
- refactors 08-patch-article-by-article-id - added 200 test for when inc_votes is positive as well as negative / refactored SQL query in updateArticleVotesById to edit votes within one query rather than two.

Thanks

### CODE REVIEW

File: __tests__/integration.test.js [hunk 1]
```diff
@@ -104,6 +104,46 @@ describe('GET /api/articles', () => {
...
+    test('status 200: returns an empty array if given a topic that has no articles', () => {
+        return request(app)
+        .get("/api/articles?topic=paper")
+        .expect(200)
+        .then(({body}) => {
+            const articles = body.articles
+            expect(articles).toHaveLength(0)
+        })
+    });
+    test('status 404: returns topic not found when querying a topic that doesnt exist', () => {
```
Comment (nc-poppy-m) [inline __tests__/integration.test.js:139]: Excellent testing

File: controllers/articles.controllers.js [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
+const { fetchTopics } = require("../models/topics.models")
 
 exports.getArticles = (req, res, next) => {
-    fetchArticles()
-        .then((articles) => {
-            res.status(200).send({ articles })
-        })
-        .catch((err) => next(err))
+    const { topic } = req.query
+    Promise.all([fetchTopics(topic), fetchArticles(topic)])
```
Comment (nc-poppy-m) [inline controllers/articles.controllers.js:7]: Lovely use of promise.all!

**Review by nc-poppy-m — COMMENTED** [review]
Great work!

**Comment by EliR94** [comment]
thank you so much for that lovely feedback @nc-poppy-m 🥳

**Pull request merged**